src/compiler/crystal/codegen/codegen.cr

```diff
@@ -1186,6 +1186,10 @@ module Crystal
       read_instance_var node.type, node.obj.type, node.name, @last
     end
 
+    def read_instance_var(node_type, type : TypeDefType, name, value)
+      read_instance_var(node_type, type.typedef, name, value)
+    end
```

Comment:
This is great! However, it might be a bit simpler to call remove_typedef on the other read_instance_var definition (that's how it's done everywhere else).